zebra/zebra_nhg.c

```diff
@@ -100,6 +100,16 @@ static void nexthop_set_resolved(afi_t afi, const struct nexthop *newhop,
 				   newhop->nh_label->num_labels,
 				   &newhop->nh_label->label[0]);
 
+	/* Append labels of parent as well.
+	 * We have to do this since nexthop objects being
+	 * installed need all of them. We can't iterate up the tree
+	 * when we install the route like we used to.
+	 */
```

Comment:
I guess I'm still missing something - maybe this comment is throwing me off - why can't we iterate as we ... always have done? We have the recursive nexthop that is being resolved, and we have the resolving nexthops - what is it that we can't iterate over?